### PR TITLE
Issue #530: Added backwards-compatible wrapper function for system_get_date_formats().

### LINKED ISSUE #530: [DX] Add bc layer for system_get_date_types()
In Drupal 7 the function was named `system_get_date_types()` in Backdrop it's called `system_get_date_formats`.  Can we add a backwards-compatibility wrapper function for this into drupal.inc? 

Reported in http://enzolutions.com/articles/2014/12/29/backdrop-cms-first-impressions

### PR DESCRIPTION
**Pull request by ghost**
Fixes https://github.com/backdrop/backdrop-issues/issues/530.

### CODE REVIEW

**Comment by quicksketch** [comment]
Since this isn't an obvious Drupal backwards-compatibility function, we should include a `@deprecated` line in the docblock. See the other functions in that file for reference.

**Comment by ghost** [comment]
@quicksketch I've updated the commit to include the 'deprecated' line.

**Comment by quicksketch** [comment]
Looks great, thanks!

File: core/includes/drupal.inc [hunk 1]
```diff
@@ -889,5 +889,39 @@ function system_settings_form_submit($form, &$form_state) {
...
+ * Alters Backdrop's list of defined date formats and attributes to return the
+ * expected date types and attributes for Drupal's system_get_date_types()
+ * function.
+ *
+ * @deprecated since 1.0
+ */
+function system_get_date_types($type = NULL) {
+  $formats = system_get_date_formats($type);
+
+  if ($type) {
```
Comment (davereid) [inline core/includes/drupal.inc:901]: This should be !$type ??

Comment (quicksketch) [inline core/includes/drupal.inc:901]: Hi @davereid! I think you're right. And goodness this function's logic is confusing to follow! Let's open a new issue to fix it.

**Pull request merged**